pallets/poa/src/lib.rs

```diff
@@ -134,29 +260,107 @@ decl_module! {
+        }
+
+        /// Withdraw from treasury. Only Master is allowed to withdraw
+        #[weight = (0, Pays::No)]
+        pub fn withdraw_from_treasury(origin, recipient: T::AccountId, amount: BalanceOf<T>) -> dispatch::DispatchResult {
+            ensure_root(origin)?;
+            Self::withdraw_from_treasury_(recipient, amount)
+        }
+
+        /// Enable/disable emission rewards by calling this function with true or false respectively.
+        /// Only Master can call this.
+        #[weight = (0, Pays::No)]
+        pub fn set_emission_status(origin, status: bool) -> dispatch::DispatchResult {
+            ensure_root(origin)?;
+            EmissionStatus::put(status);
+            Ok(())
+        }
+
+        /// Set the minimum number of slots in the epoch, i.e. storage item MinEpochLength. This
+        /// does not effect the epoch in progress but only subsequent epochs
+        #[weight = (0, Pays::No)]
+        pub fn set_min_epoch_length(origin, length: EpochLen) -> dispatch::DispatchResult {
+            ensure_root(origin)?;
+            ensure!(length > 0, Error::<T>::EpochLengthCannotBe0);
+            MinEpochLengthTentative::put(length);
+            Ok(())
+        }
+
+        /// Set the maximum number of active validators, i.e. storage item MaxActiveValidators. This
+        /// does not effect the epoch in progress but only subsequent epochs
+        #[weight = (0, Pays::No)]
+        pub fn set_max_active_validators(origin, count: u8) -> dispatch::DispatchResult {
+            ensure_root(origin)?;
+            ensure!(count > 0, Error::<T>::NeedAtLeast1Validator);
+            MaxActiveValidatorsTentative::put(count);
+            Ok(())
+        }
+
+        /// Set the maximum emission rewards per validator per epoch.
+        #[weight = (0, Pays::No)]
+        pub fn set_max_emm_validator_epoch(origin, emission: u128) -> dispatch::DispatchResult {
+            ensure_root(origin)?;
+            <MaxEmmValidatorEpoch<T>>::put(emission.saturated_into::<BalanceOf<T>>());
+            Ok(())
+        }
+
+        /// Set percentage of emission rewards locked per epoch for validators
+        #[weight = (0, Pays::No)]
+        pub fn set_validator_reward_lock_pc(origin, lock_pc: u8) -> dispatch::DispatchResult {
+            ensure_root(origin)?;
+            ensure!(
+                lock_pc <= 100,
+                Error::<T>::PercentageGreaterThan100
+            );
+            ValidatorRewardsLockPercent::put(lock_pc);
+            Ok(())
+        }
+
+        /// Set percentage of emission rewards for treasury in each epoch
+        #[weight = (0, Pays::No)]
+        pub fn set_treasury_reward_pc(origin, reward_pc: u8) -> dispatch::DispatchResult {
+            ensure_root(origin)?;
+            ensure!(
+                reward_pc <= 100,
+                Error::<T>::PercentageGreaterThan100
+            );
+            TreasuryRewardsPercent::put(reward_pc);
+            Ok(())
+        }
```

Comment:
If we need to enforce `TreasuryRewardsPercent::get() + ValidatorRewardsLockPercent::get() == 100` then these two functions can be merged into a single atomic action.

```suggestion
        /// Set percentage of emission rewards locked per epoch for validators and percentage of emission rewards
        /// for treasury per epoch.
        #[weight = (0, Pays::No)]
        pub fn set_emission_rewards(origin, lock_pc: u8, treasury_pc: u8) -> dispatch::DispatchResult {
            ensure_root(origin)?;
            ensure!(
                lock_pc.saturating_add(treasury_pc) == 100,
                Error::<T>::PercentageTotalNot100
            );
            ValidatorRewardsLockPercent::put(lock_pc);
            TreasuryRewardsPercent::put(treasury_pc);
            Ok(())
        }
```

I don't consider this a serious issue. Only a problem if Master makes mistake, or if setting the two values happens to occur over and epoch boundary.

Reply:
`TreasuryRewardsPercent` and `ValidatorRewardsLockPercent` are independent and don't need to add up to 100.